Lib/asyncio/__init__.py

```diff
@@ -43,3 +44,21 @@
 else:
     from .unix_events import *  # pragma: no cover
     __all__ += unix_events.__all__
+
+
+__all__ += ('StreamReader', 'StreamWriter')  # deprecated
+
+
+def __getattr__(name):
+    if name == 'StreamReader':
+        warnings.warn("StreamReader is deprecated, use asyncio.Stream instead",
+                      DeprecationWarning,
+                      stacklevel=2)
+        return Stream
```

Comment:
Do you think it's a good idea to return `Stream` when the user has *explicitly* asked for a `StreamReader`?  Is there a way we can return `StreamReader` and `StreamWriter` (at least in 3.8)?  I feel that would be a smoother migration path.

Reply:
The real question is that: why user need to import `Stream` class at all?
I guess for two reasons: hacking by deriving a custom stream class from `StreamReader` for example. Sorry, this way is broken, `Stream` has no compatible constructor with `StreamReader`.
Note, both `Stream.__init__` and `StreamReader.__init__` are private constructors, a user should never call them directly but use factories like `open_connection()` / `connect()`.
The second reason is `isinstance(obj, StreamReader)` check. 
Now both `isinstance(obj, asyncio.Stream)` and `isinstance(obj, asyncio.StreamReader)` returns the same result, which is good I think.

Reply:
> I guess for two reasons: hacking by deriving a custom stream class from StreamReader for example. Sorry, this way is broken, Stream has no compatible constructor with StreamReader.

Right, but it means you are making a backwards incompatible change in 3.8 by returning `Stream`.  We can't do that.